src/coreclr/jit/importer.cpp

```diff
@@ -3671,9 +3672,13 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
         // For mismatched VM (AltJit) we want to check all methods as intrinsic to ensure
         // we get more accurate codegen. This particularly applies to HWIntrinsic usage
         assert(!info.compMatchedVM);
-    }
 
-    ni = lookupNamedIntrinsic(method);
+        if (ni == NI_Illegal)
+        {
+            // Early exit for the common AltJit scenario
```

Comment:
Do we really need this for altjit when it is handled below anyway?

Reply:
Its an early exit that avoids 5-7 additional checks, which will be done for all methods from all assemblies.

Reply:
I think the fewer cases are special-cased due to altjit, the better, even if it makes altjit slower.

Reply:
Removed the early exit check